runtime/libc/wasi_sdk_backing.c

```diff
@@ -351,74 +200,189 @@ static wasi_errno_t wasi_fromerrno(int errno_) {
     return 0;
 }
 
+__attribute__((noreturn))
+void wasi_unsupported_syscall(const char *syscall) {
+    fprintf(stderr, "Syscall %s is not supported\n", syscall);
+    exit(EXIT_FAILURE);
+}
+
 /**
- * @brief Used by a WASI module to copy the argument buffer into linear memory write an 
+ * @brief Returns arguments buffer into linear memory write an 
  * indirect to the base offset of the buffer 
  * 
- * @param argv_p_offset 
- * @param argv_buf_offset 
- * @return i32 
+ * @param argv_ptr
+ * @param argv_buf_ptr
+ * @return WASI_ESUCCESS
  */
-wasi_errno_t wasi_snapshot_preview1_args_get(u32 argv_p_offset, u32 argv_buf_offset){
-    char *argv_buf = get_memory_ptr_for_runtime(argv_buf_offset, runtime_argv_buffer_len);
+wasi_errno_t wasi_snapshot_preview1_args_get(
+    wasi_size_t argv_retptr, 
+    wasi_size_t argv_buf_retptr
+){
+    wasi_size_t *argv = (wasi_size_t *)get_memory_ptr_for_runtime(argv_retptr, sizeof(wasi_size_t) * runtime_argc);
+    char *argv_buf = get_memory_ptr_for_runtime(argv_buf_retptr, runtime_argv_buffer_len);
```

Comment:
Maybe add an issue for the future? Options include 1. returning an error (likely reasonable), 2. bombing out either by panicing, or using longjmp to return from the module.

Reply:
hmmmm, did this comment become a "dangling comment". Was referencing the TODO below. Sorry, ignore this.

Reply:
I'll add this to an issue for improved bound checking behavior.

Reply:
I rephrased this issue a bit after rereading the section on "traps" in the WebAssembly documentation. Given how this is phrased, I think setjmp/longjmp into a stubbed trap handler is probably the most future-proof solution. https://github.com/gwsystems/aWsm/issues/43